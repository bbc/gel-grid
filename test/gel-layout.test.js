const sass = require('sass');

describe('gel-layout()', () => {
  it('creates a layout to wrap around items', async () => {
    const { css } = await sass.compileAsync('test/data/gel-layout/t1.scss', { quietDeps: true, logger: sass.Logger.silent });

    expect(css).toMatchSnapshot();
  });

  it('creates a layout to wrap around items, with an optional whitespace adjustment', async () => {
    const { css } = await sass.compileAsync('test/data/gel-layout/t2.scss', { quietDeps: true, logger: sass.Logger.silent });

    expect(css).toMatchSnapshot();
  });
});
