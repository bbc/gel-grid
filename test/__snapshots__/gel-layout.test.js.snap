// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gel-layout() creates a layout to wrap around items 1`] = `
".gel-layout {
  direction: ltr;
  display: flex;
  flex-flow: row wrap;
  flex-grow: 1;
  list-style: none;
  margin-right: 0;
  margin-left: -8px;
  padding-right: 0;
  padding-left: 0;
  text-align: left;
}
@media (min-width: 37.5em) {
  .gel-layout {
    margin-left: -16px;
  }
}"
`;

exports[`gel-layout() creates a layout to wrap around items, with an optional whitespace adjustment 1`] = `
".gel-layout {
  direction: ltr;
  display: flex;
  flex-flow: row wrap;
  flex-grow: 1;
  list-style: none;
  margin-right: 0;
  margin-left: -8px;
  padding-right: 0;
  padding-left: 0;
  text-align: left;
  letter-spacing: -0.31em;
  text-rendering: optimizespeed;
}
.gel-layout {
  word-spacing: -0.43em;
}

@media (min-width: 37.5em) {
  .gel-layout {
    margin-left: -16px;
  }
}"
`;
